### PR TITLE
[LLVM] Do not require shell for some tests

### PR DESCRIPTION
**Pull request by jayfoad**
Remove `REQUIRES: shell` from some tests that seem fine without it.
Tested on Windows and with LIT_USE_INTERNAL_SHELL=1 on Linux.

### CODE REVIEW

**Review by MaskRay — APPROVED** [review]

File: llvm/test/tools/llvm-cov/gcov/intermediate-format.test [hunk 1]
```diff
@@ -1,4 +1,3 @@
-REQUIRES: shell
```
Comment (MaskRay) [inline llvm/test/tools/llvm-cov/gcov/intermediate-format.test:1]: If `REQIRES: shell` is the first line, delete the blank line after it as well.

File: llvm/test/tools/llvm-rc/windres-prefix.test [hunk 2]
```diff
@@ -1,4 +1,3 @@
-; REQUIRES: shell
```
Comment (MaskRay) [inline llvm/test/tools/llvm-rc/windres-prefix.test:1]: If REQIRES: shell is the first line, delete the blank line after it as well.

**Comment by jayfoad** [comment]
Hi @zmodem, I have just noticed that you have reverted some of these changes because of gnuwin32 failures. What did the failures look like? Is there a bot I can look at? I'd prefer to find a more precise way to disable them than `REQUIRES: shell` because these tests clearly do not require an external shell (at least not on all platforms).

**Comment by zmodem** [comment]
The ones with `ln` are pretty simple :-)

```
'ln': command not found
```

`Other/lit-unicode.txt` is less straight-forward:

```
UnicodeEncodeError: 'charmap' codec can't encode characters in position 6-9: character maps to <undefined>
```

I assumed that's some gnuwin32 thing having poor unicode support.

Both of these can be seen here https://logs.chromium.org/logs/chromium/buildbucket/cr-buildbucket/8745531620523249025/+/u/package_clang/stdout

**Comment by pogo59** [comment]
Yeah I think `ln` qualifies as deserving a `REQUIRES: shell` ?

For the unicode thing, seems sketchy to expect it to work on Windows. I cut the file down to just the `echo` command and tried to run it on my box, and got junk out.

**Comment by petrhosek** [comment]
`ln` should be invoked by the internal shell as an external command, the error suggests that the internal shell may have an issue with resolving the right tool.

**Comment by jayfoad** [comment]
> Yeah I think `ln` qualifies as deserving a `REQUIRES: shell` ?

I don't think so. `REQUIRES: shell` means the RUN lines need to be run using bash instead of lit's internal shell, but (as @petrhosek said) the internal shell can still run external commands like /usr/bin/ln.

I think `UNSUPPORTED: system-windows` is probably more appropriate on the grounds that Windows doesn't have Unix-like symlinks.

> For the unicode thing, seems sketchy to expect it to work on Windows. I cut the file down to just the `echo` command and tried to run it on my box, and got junk out.

This sounds similar to the printf problems that were worked around in #83907, and again the solution was `UNSUPPORTED: system-windows`.

**Pull request merged**